conn/i2c/i2c.go

```diff
@@ -83,39 +85,123 @@ func (d *Dev) Write(b []byte) (int, error) {
 	return len(b), nil
 }
 
-// ReadReg writes the register number to the I²C bus, then reads data.
-//
-// It's a wrapper for Tx()
-func (d *Dev) ReadReg(reg byte, b []byte) error {
+// DevReg is a Dev that exposes memory mapped registers.
+type DevReg struct {
+	Dev
+	// Order specifies the binary encoding of words. It is expected to be either
+	// binary.BigEndian or binary.LittleEndian or a specialized implemented if
+	// necessary. A good example of such need is devices communicating 32bits
+	// little endian values encoded over 16bits big endian words.
+	Order binary.ByteOrder
+}
+
+// ReadRegRaw writes the register number to the I²C bus, then reads data
+// unmarshalled.
+func (d *DevReg) ReadRegRaw(reg byte, b []byte) error {
 	return d.Tx([]byte{reg}, b)
 }
 
-// Shortcuts
+// ReadStruct writes the register number to the I²C bus, then reads data
+// marshalled.
```

Comment:
It would be nice to have more comments here. What is the intent? Size of struct = number of bytes read. Mention that binary.Read is used to read into struct and that it applies the byte ordering.

Perhaps an example could really help too!